components/private/XCTest/src/XCTest+MaterialColor.h

```diff
@@ -0,0 +1,21 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <XCTest/XCTest.h>
+
+@interface XCTestCase (MaterialColor)
+
+- (void)assetEqualFirstColor:(UIColor *)firstColor secondColor:(UIColor *)secondColor;
```

Comment:
Xcode's assertions are macros so that failures jump you directly to the failing test case. Can we implement this method as a macro instead?

Reply:
I made a change in fbd7964f7bab204a254d00d8e2690c503cf53ebd to address the macro to match the style. I thought about directly use whole macro, but I am not sure if it could be correctly ported to swift since it is not a simple marco. I can keep both the method and make the marco to duplicate the functions, so it works under objective-c for the jump function, but it is not going to reuse the code. How do you feel?

Is `MDCAssertEqualColor` named for namespace concerns? I used `XCTAssertEqualColors`, which I haven't taken it into consideration.

Reply:
The macro you added won't change the resulting behavior of calling these APIs because it's still calling a separate method altogether.

I found the following article which describes how to wire up a custom assertion such that the failures appear at the expected call site: https://www.bignerdranch.com/blog/creating-a-custom-xctest-assertion/

Reply:
I just tried it, the original implementation (method called in macro) actually takes me to the correct failing case (though not the line). It doesn't bring me to the file with the custom assertion method.

The reason why the method is used in macro is for reusability (the behavior is different), but keeping a method there is good for it being used in swift.

Reply:
Rad! It sounds like we don't need a macro either then. May still be be preferable for consistency with other test assertions though, will leave that up to you!